File: 1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
         Stack<Character> stk = new Stack<>();
        for (char i : s.toCharArray()) {
            if (!stk.isEmpty() && Character.toLowerCase(i) == Character.toLowerCase(stk.peek()) &&
                    (i != stk.peek())) {
                stk.pop();
            } else {
                stk.push(i);
            }
        }
        StringBuilder ans = new StringBuilder();
        while (!stk.isEmpty()) {
            ans.insert(0, stk.pop());
        }
        return ans.toString();
    }
}

File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        stack<char>stk;
        for(char i:s){
            if(!stk.empty()&&char(tolower(i))==char(tolower(stk.top())) &&(int(i)!=int(stk.top()))){stk.pop();}
            else stk.push(i);
        }
        string ans="";
        while(!stk.empty()){ans=stk.top()+ans;stk.pop();}
        return ans;
    }
};